hasktorch/src/Torch/Functional.hs

```diff
@@ -943,3 +943,26 @@ flattenAll
   -> Tensor -- ^ output
 flattenAll t =
   unsafePerformIO $ (cast3 ATen.flatten_tll) t (0 :: Int) (-1 :: Int)
+
+-- | smoothL1Loss
+smoothL1Loss
+  :: Reduction -- ^ reduction
+  -> Tensor -- ^ target
+  -> Tensor -- ^ input
```

Comment:
Most other losses take `input` before the `target`, so it's best not to change the convention.

Reply:
Yeah, makes sense.

Reply:
@apaszke Do you think, we should rather change others, since that would also satisfy the currying principle?

Reply:
I think that in most cases loss functions are applied together to the input and the target, so there's usually no currying involved, and this order is more natural I think. I'd keep it as it used to be.

Reply:
Roger that. :)